### DISCUSSION TITLE
Custom Audio Tracks

### DISCUSSION THREAD

**Discussion by codercms** [vidstack/player#853] [Q&A]
I have a few questions about player state management in case of audio tracks:

* Is it possible to set custom labels for audio tracks? 
E.g. when there are many audio tracks in HLS stream and they're named like this: eng0, eng1, eng2
But actual audio track names could be fetched from server, e.g. eng0 => Some voice-over actors

* Is it possible to add custom audio tracks?
E.g. when there are separate HLS streams for each audio track

**Comment by mihar-22** [comment]
> Is it possible to set custom labels for audio tracks?

Not at the moment. Ideally, it should be set in your HLS playlist. We can definitely consider exposing some API to do it though, let me know if you have an idea.

> Is it possible to add custom audio tracks?

Not sure what you exactly mean here, can you elaborate? Do you mean add custom audio tracks outside of a HLS playlist to the player? If so, that is currently not possible.

**Reply by codercms** [reply]
> let me know if you have an idea.

I wonder is it possible to make some hook to pre-process audio tracks before they're stored in player state?

Or just having something more general:
```ts
// update player store like svelte store

const mapping = {
   0: "English";
   1: "Spanish"
};

player.$store.update((prevState) => {
    prevState.audioTracks = prevState.audioTracks.map((audioTrack) => {
           if (mapping[audioTrack.id]) {
               audioTrack.label = mapping[audioTrack.id];
           }

           return audioTrack;
    });

    return prevState;
});
```

> Not sure what you exactly mean here, can you elaborate?

I mean where there are actually virtual audio tracks (to change audio track you have to load different HLS stream):
Deutsch translation - HLS Stream 1 (http://cdn.host/streams/stream-de.m3u8)
English translation   - HLS Stream 2 (http://cdn.host/streams/stream-en.m3u8)
Spanish translation  - HLS Stream 3 (http://cdn.host/streams/stream-sp.m3u8)

I guess it should be possible to do it by specifying audio tracks as different sources somehow , e.g.:
```html
  <media-outlet>
    <source type="virtual-audio-track" src="//cdn.host/hls-stream-1.m3u8" label="English" />
    <source type="virtual-audio-track" src="//cdn.host/hls-stream-2.m3u8" label="Spanish" />
  </media-outlet>
```
